python/ray/autoscaler/azure/node_provider.py

```diff
@@ -173,6 +180,7 @@ def create_node(self, node_config, tags, count):
         config["tags"] = config_tags
         config["location"] = location
         name_tag = config_tags.get(TAG_RAY_NODE_NAME, "node")
+        node_type = config_tags.get(TAG_RAY_NODE_TYPE, "head")
```

Comment:
wouldn't this set all nodes to type head?

Reply:
it defaults to head if it's not there. maybe shouldn't have the default?

Reply:
yeah, maybe both of these should fail if the tags are not provided?